airgun/entities/host_new.py

```diff
@@ -272,15 +435,146 @@ def get_parameters(self, entity_name):
+            ValueError: If parameter with the same name already exists
+        """
+
+        # Sanitize input
+        parameter_type = parameter_type.lower()
+        available_types = ['string', 'boolean', 'integer', 'real', 'array', 'hash', 'yaml', 'json']
+        if parameter_type not in available_types:
+            raise ValueError(f'Parameter type {parameter_type} is not valid')
+
         view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
         view.wait_displayed()
         self.browser.plugin.ensure_page_safe()
+        view.parameters.click()
+        view.parameters.searchbar.fill(parameter_name)
+        view.wait_displayed()
+        # Check if parameter with given name does not exist. If it does, raise ValueError.
+        if view.parameters.parameters_table.row_count != 0:
+            if view.parameters.parameters_table[0][0].text == parameter_name:
+                raise ValueError(f'Parameter with name {parameter_name} already exists')
+
         view.parameters.add_parameter.click()
-        view.parameters.new_parameter_name.fill(parameter_name)
-        view.parameters.new_parameter_type.fill(parameter_type)
-        view.parameters.new_parameter_value.fill(parameter_value)
+        view.parameters.parameter_name_input.fill(parameter_name)
+        view.parameters.parameter_type_input.fill(parameter_type)
+        view.parameters.parameter_value_input.fill(parameter_value)
         view.parameters.confirm_addition.click()
 
+    def edit_parameter(
+        self,
+        entity_name,
+        parameter_to_change,
+        new_parameter_name=None,
+        new_parameter_type=None,
+        new_parameter_value=None,
+    ):
+        """
+        Function that can edit parameter name, type or value separetely or all of them at once.
+
+        Args:
+            entity_name: Name of the entity to be edited
+            parameter_to_change: Name of the parameter to be edited
+            new_parameter_name: New name of the parameter
+            new_parameter_type: New type of the parameter
+                [string, boolean, integer, real, array, hash, yaml, json]
+            new_parameter_value: New value of the parameter
+
+        Raises:
+            ValueError: No new parameter name, type or value provided
+            ValueError: If parameter type is not valid
+            ValueError: If parameter_to_change is not found on host
+            ValueError: If parameter with new name already exists
+        """
+
+        if not any([new_parameter_name, new_parameter_type, new_parameter_value]):
+            raise ValueError(
+                'At least one of the new_parameter_name, new_parameter_type, new_parameter_value '
+                'must be provided!'
+            )
+        # Sanitize input
+        new_parameter_type = new_parameter_type.lower()
+        available_types = ['string', 'boolean', 'integer', 'real', 'array', 'hash', 'yaml', 'json']
```

Comment:
could we have this defined somewhere just once, so it can be reused across functions?